nixos/lib/test-driver/test_driver/__init__.py

```diff
@@ -33,6 +33,18 @@ def __call__(self, parser, namespace, values, option_string=None):  # type: igno
         setattr(namespace, self.dest, values)
 
 
+class WriteableDir(argparse.Action):  # type: ignore
+    def __call__(self, parser, namespace, values, option_string=None):  # type: ignore
+        if not values.is_dir():
+            raise argparse.ArgumentTypeError("{0} is not a directory".format(values))
+        if os.access(values, os.W_OK):
+            setattr(namespace, self.dest, values)
+        else:
+            raise argparse.ArgumentTypeError(
+                "{0} is not a writeable directory".format(values)
+            )
```

Comment:
If we're gonna introduce this mechanism, can we also use it for `$TEMPDIR`?

Reply:
I did that, but I needed to redefine itsince it should not raise a argparse specific exception